### DISCUSSION TITLE
Single "Fader" MCU controller

### DISCUSSION THREAD

**Discussion by d1One** [tttapa/Control-Surface#669] [Q&A]
Hello. I am wondering if Control-Surface is suited to make the type of controller I want; my goal is to do something similar to:
[iCon - Platform Nano Air](https://iconproaudio.com/product/platform-nano-air/)
[PreSonus - FaderPort](https://www.presonus.com/products/FaderPort)
[Behringer - X-Touch One ](https://www.behringer.com/product.html?modelCode=P0CAP)

But much smaller and simpler, with an encoder instead of a motorized fader. From what I've seen in some videos there seems to be an issue with this sort of controller and the overall MCU limitations: If we select a track with the mouse outside of the active 8 Track Bank MCU was originally built for the controller loses sync with the DAW. 
What I am wondering is if there could be some kind of solution for this done inside of Control-Surface and what the best approach would be. (in case it's even possible)

Or an even simpler question: How do those 1-fader controllers know which track is selected on the DAW?

Thanks very much for your time.

**Comment by Veroledez** [comment]
Hola,  prueba lo siguiente, subele el programa de ingenieria inversa a tu arduino, configuralo como mackie    control en tu daw, luego ve que imprime en el monitor serie cuando seleccionas un track que  no esta en el banco de mackie actual, luego usas ese dato obtenido para actualizar el valor del codificador.

**Reply by d1One** [reply]
Hello. 
Thank you for your help! Regardless of what track I activate in Ableton I always get the same output in serial:
```
90 19 7f (SELECT_2) on cable 1
90 1a 00 (SELECT_3) on cable 1
90 1b 00 (SELECT_4) on cable 1
90 1c 00 (SELECT_5) on cable 1
90 1d 00 (SELECT_6) on cable 1
90 1e 00 (SELECT_7) on cable 1
90 1f 00 (SELECT_8) on cable 1
90 18 00 (SELECT_1) on cable 1
```
There does not seem to be any special message that tells the MCU which channel/track is active, this is why I am confused about how those 1 fader controllers I listed above even work and how I would approach emulating one with Control-Surface. I am only interested in having the LCD info follow the DAW, everything else is easy I think.

**Reply by Veroledez** [reply]
Creo que tendrias que modificar el script de mcu para que cuando se seleccione un track fuera del bank mcu obtenga el valor y lo envie al controlador, y este lo guarde en una variable en donde pueda sumar y restar segun    el sentido de giro
